sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CollationTypeCasts.scala

```diff
@@ -52,6 +52,14 @@ object CollationTypeCasts extends TypeCoercionRule {
       overlay.withNewChildren(collateToSingleType(Seq(overlay.input, overlay.replace))
         ++ Seq(overlay.pos, overlay.len))
 
+    case stringRPad: StringRPad =>
+      val Seq(str, pad) = collateToSingleType(Seq(stringRPad.str, stringRPad.pad))
+      stringRPad.withNewChildren(Seq(str, stringRPad.len, pad))
+
+    case stringLPad: StringLPad =>
+      val Seq(str, pad) = collateToSingleType(Seq(stringLPad.str, stringLPad.pad))
+      stringLPad.withNewChildren(Seq(str, stringLPad.len, pad))
```

Comment:
it would be good if we could combine these, like you did before

try if something like this works:
```
      val Seq(str, len, pad) = stringPadExpr.children
      val Seq(newStr, newPad) = collateToSingleType(Seq(str, pad))
      stringPadExpr.withNewChildren(Seq(newStr, len, newPad))
```

Reply:
@uros-db  done. please re-review.